odoo/addons/base/models/res_currency.py

```diff
@@ -52,6 +52,23 @@ class Currency(models.Model):
         ('rounding_gt_zero', 'CHECK (rounding>0)', 'The rounding factor must be greater than 0!')
     ]
 
+    def write(self, vals):
+        res = super().write(vals)
+        if 'active' not in vals:
+            return res
+        # Automatically activate group_multi_currency if there is more than 1 active currency, deactivate it otherwise
+        group_user = self.env.ref('base.group_user', raise_if_not_found=False)
+        group_mc = self.env.ref('base.group_multi_currency', raise_if_not_found=False)
+        group_pp = self.env.ref('product.group_product_pricelist', raise_if_not_found=False)
+        active_currency_count = self.search_count([('active', '=', True)])
+        if group_user and group_mc and active_currency_count > 1:
+            group_user._apply_group(group_mc)
+            if group_pp:
+                group_user._apply_group(group_pp)
```

Comment:
price lists are managed in module product. Since sales orders use the price list feature for multi currency, we activate the basic price list group `group_product_pricelist` here

Reply:
this no longer applies as there is now a hook method in `res.config.settings`  called `_toggle_group_multi_currency` and this is overridden in module `product`